meshnets/utils/datasets.py

```diff
@@ -1,86 +1,113 @@
+            mesh.vtk
+            graph.pt
+        sample_3/
+            mesh.vtk
+            graph.pt
+
+    Note that the directory, file names, and file extensions can be different.
+    `process_data` allows to produce the graph files upon instanciating the 
+    dataset class from a folder containing only the mesh files.
+
     This class inherits from torch_geometric Dataset and implements its
-    abstract methods `len` and `get` from the FromDiskDataset class.
-    This offers acces to the torch_geometric Dataset attributes when the .pt
-    files retrieved from disk are torch_geometric Data.
+    abstract methods `len` and `get`. This offers acces to the torch_geometric
+    Dataset attributes when the .pt files retrieved from disk are
+    torch_geometric Data.
 
     It also offer a new property called 'num_label_features' following
     the implementation of 'num_node_features' and 'num_edge_features'
     in torch_geometric Dataset.
     """
 
-    def __init__(self, data_dir, *args, **kwargs):
+    def __init__(self,
+                 data_dir: str,
+                 *args,
+                 process_data: bool = False,
+                 mesh_file_ext: str = '.vtk',
+                 graph_file_ext: str = '.pt',
+                 **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        self.from_disk_dataset = FromDiskDataset(data_dir=data_dir)
+
+        self.root_dir = data_dir
+        self.samples = sorted(os.listdir(data_dir))
+
+        self.mesh_file_ext = mesh_file_ext
+        self.graph_file_ext = graph_file_ext
+
+        if process_data:
+            self.process_data()
+
+    def process_data(self) -> None:
+        """Process the mesh files in the data directory to their graph
+        representation. Save the graph file in the same sample directory."""
+        for i in range(self.len()):
+            mesh_file_path = self.get_mesh_path(i)
+
+            processed_graph = data_processing.mesh_file_to_graph_data(
+                mesh_file_path, WIND_VECTOR, load_pressure=True)
+
+            processed_file_path = Path(mesh_file_path).with_suffix(
+                self.graph_file_ext)
+
+            torch.save(processed_graph, processed_file_path)
 
     def len(self) -> int:
-        """Returns the number of graphs stored in the dataset.
-        
-        This is the number of files in the dataset from disk."""
-        return len(self.from_disk_dataset)
+        """Return the number of samples in the dataset."""
+        return len(self.samples)
+
+    def get_sample_path(self, idx: int) -> str:
+        """Get the path to the sample directory at the given index."""
+        return os.path.join(self.root_dir, self.samples[idx])
+
+    def get_mesh_path(self, idx: int) -> str:
+        """Get the path to the mesh file at the given index."""
+        sample_path = self.get_sample_path(idx)
+        mesh_path = glob.glob(
+            os.path.join(sample_path, f'*{self.mesh_file_ext}'))[0]
```

Comment:
This will not work if you have other vtk files in `sample_path`, which is very likely... Should we set the expected names of the vtk files?

Reply:
I access here the first vtk file found in the sample dir, if there are multiple the behavior it will probably access a random one. 

What do you suggest? I don't know what why there would be multiple .vtk files in a single sample dir.

Reply:
That could be an option yes, I can easily replace self.mesh_file_ext by self.mesh_file_name, and similarly for the graph file name. This would allow multiple .vtk and .pt files for the same sample, allowing also for different graph representation in the same data dir.

This comes at the cost of having to define the file names but it might be worth it. Let me know if I should go for this option.

Reply:
I'd go with this option, yes. Thanks @VictorGillioz!

Reply:
I added a warning when the path to a file that does not exist is returned. I can also use assert, throw an error, or do nothing. Do you know best practice in this case?